### PR TITLE
Recording enhancements

### PR DESCRIPTION
**Pull request by emveepee**
Recording changes

I. General

- reload recordings and reset modification time when a settings change impacts recordings
- delete recording channel json cache to allow for changes outside of Jellyfin.  JF only calls for new recordings when the database entries are not cached
- use episode mode without S/E for all non movie recordings, Jellyfin was using the subtitle as title and not displaying the title on clips.
- parse duplicated S/E information from the sub-title passed in the API.
- change cache key, add-on was using a key that changed often, leaving orphan JSON in the cache for every recording

II. Recording polling

In 10.9 Jellyfin was re-enabling disabled plugin causing user performance issues (not sure why though)

- don't poll for recording until connected
- reduce initial connection time to 5 seconds to avoid overlapping connection attempts
- allow configuration of polling interval, the more frequent the more responsive the addon is to changes outside Jellyfin
- catch connection errors in the plugin, don't throw errors to core.

III. In-progress recordings problems

a) In-progress recordings do not play when recordings are flagged as in-progress

- default in-progress to direct play.
- append growing flag to recordings to not send content-length.

b) No resume while in-progress

- don't know what to do to make this work.

c) Slow playback on initial in-progress recordings

- cannot change the analyzeduration so FFprobe waits for 200MB before playback
- if users give up, orphan FFmpeg tasks could be left running.
- don't know what to do to make this work.

IV. Security changes

- configure IPv4 so FFmpeg doesn't use IPv6 when a hostname is specified
- adapt to NextPVR unauthenticated access fix starting in 6.1.6
- allow unauthenticated recording streaming so clients can direct play.  Jellyfin does not proxy client so security was failing and only transcoded play was available.

### CODE REVIEW

**Comment by emveepee** [comment]
@crobibero note that there are some things in this patch that don't fix in-progress recordings so any suggestions would be helpful before merging.

File: Jellyfin.Plugin.NextPVR/Jellyfin.Plugin.NextPVR.csproj [hunk 1]
```diff
@@ -17,7 +18,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Jellyfin.Controller" Version="10.*-*" />
+    <PackageReference Include="Jellyfin.Controller" Version="10.9.6" />
```
Comment (crobibero) [inline Jellyfin.Plugin.NextPVR/Jellyfin.Plugin.NextPVR.csproj:21]: Please revert

File: Jellyfin.Plugin.NextPVR/LiveTvService.cs [hunk 2]
```diff
@@ -92,47 +117,83 @@ public async Task EnsureConnectionAsync(CancellationToken cancellationToken)
...
+        httpClient.Timeout = TimeSpan.FromSeconds(5);
+        bool updateConfiguration = false;
+        bool validConfiguration = false;
+        if (!string.IsNullOrEmpty(Plugin.Instance.Configuration.StoredSid) && !string.IsNullOrEmpty(Plugin.Instance.Configuration.CurrentWebServiceURL) )
         {
             string request = $"{baseUrl}/service?method=session.valid&device=jellyfin&sid={Plugin.Instance.Configuration.StoredSid}";
+            var httpClient1 = new HttpClient();
+            // using HttpRequestMessage request1 = new(HttpMethod.Options, request);
+            HttpResponseMessage response = await httpClient1.GetAsync(request, cancellationToken);
+            var b = response.RequestMessage;
```
Comment (crobibero) [inline Jellyfin.Plugin.NextPVR/LiveTvService.cs:131]: This looks like debug code?

File: Jellyfin.Plugin.NextPVR/Jellyfin.Plugin.NextPVR.csproj (hunk 1 above)
Comment (emveepee) [reply Jellyfin.Plugin.NextPVR/Jellyfin.Plugin.NextPVR.csproj]: OK I have to watch for this, VS did this automatically.

File: Jellyfin.Plugin.NextPVR/LiveTvService.cs (hunk 2 above)
Comment (emveepee) [reply Jellyfin.Plugin.NextPVR/LiveTvService.cs]: Indeed it was.

**Pull request merged**